File: src/main/java/com/example/cookblog/dto/requests/CommentRecipeRequest.java
package com.example.cookblog.dto.requests;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Builder;
import lombok.extern.jackson.Jacksonized;

@Builder
@Jacksonized
public record CommentRecipeRequest(
        @NotBlank @Size(max = 255) String author,
        @NotBlank @Size(max = 2000) String content
) {

}

File: cookblog-frontend/src/app/api/requests/comment-recipe.request.ts
import { Request } from './request';

export class CommentRecipeRequest implements Request {
  private readonly _author: string;
  private readonly _content: string;

  constructor(author: string, content: string) {
    this._author = author;
    this._content = content;
  }

  get author(): string {
    return this._author;
  }

  get content(): string {
    return this._content;
  }

  toRequestBody() {
    return {
      author: this._author,
      content: this._content,
    };
  }
}
